### ISSUE TITLE
Translatable strings in Lua with trailing spaces get stripped

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#8646]
### Game and System Information

Introduced in #8577 (master) and #8621 (1.18). Reproducible in 1.18.0+dev at 72a102ccd427caede35038bed4cd6ebebc23b29a.

### Description of the bug

When running a pot-update on Lua files, strings such as `HP: ` will appear in the .pot file as `HP:` without the trailing space. Since it doesn't match the source text, the translations can't work.
https://github.com/wesnoth/wesnoth/blob/c1b331c62fd1c4d14d83da5bb88592bb7ce518ce/data/campaigns/Secrets_of_the_Ancients/utils/zombie_recruit_dialog.lua#L23

What's happened here is that #8577 and #8621 fixed a warning in the Python code parsing those strings, without realising that the intended behavior was a bug. Reverting those stops the problem with the translations, but it's probably better to clean up the Python code.

### Steps to reproduce the behavior

1. `scons pot-update`
2. `git diff po/wesnoth-sota/wesnoth-sota.pot`

**Comment by gfgtdf** [comment]
Are there any translations that would want to remove the whitespace there? If not, we could also make a policy not to end translatable strings with a whitespace (and to put the whitespace in the untransltable part),

**Comment by stevecotton** [comment]
Not sure about the whitespace, but Arabic will probably want to put the headings and values right-to-left, so the answer is probably to make it use variable interpolation.

**Comment by Pentarctagon** [comment]
Would the fix for this be to just remove the previously buggy code?

**Comment by stevecotton** [comment]
Probably just code removal, yes; haven't reviewed what it's intended to do.

**Comment by stevecotton** [comment]
How to mark up translatable strings in Lua, none which contain a newline in when parsed by the Lua interpreter, because [when the opening long bracket is immediately followed by a newline, the newline is not included in the string.](https://www.lua.org/manual/5.4/manual.html#3.1)

```
style2 = _ 'single-quoted string'
style1 = _ "double-quoted string"
style3 = _ [==[
long-bracketed string]==]
```

The numbers in those might not exactly correspond to what's in wmlxgettext's code, still working that out.

**Comment by stevecotton** [comment]
Fixed in 1.18 by #8672, fixed in master by #8678.
